modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java

```diff
@@ -50,49 +49,38 @@
  * optimized by Tayeb Chlyah
  */
 @AllArgsConstructor
-public class CouchbaseContainer extends GenericContainer<CouchbaseContainer> {
+public class CouchbaseContainer <SELF extends CouchbaseContainer<SELF>> extends GenericContainer<SELF> {
```

Comment:
CouchbaseContainer is a final container and doesn't need SELF typing. Any good reason to add it back?

Reply:
Another reason is consistency. A lot of containers ins the modules project provide exactly the same style.

Reply:
the problem is that some languages (*ahem* Kotlin *ahem*) really suck at handling such types :D 

This is why we decided to use self-typing only on base classes and not the final ones (new modules have that)

Also, Testcontainers 2.0 will have a new, generic-less DSL

Reply:
So you suggest to revert that change?

And using `return self();` is still okay?

Reply:
Yes, please.

`return self()` is okay :)